cmd/ocp-build-data-enforcer/main_test.go

```diff
@@ -0,0 +1,104 @@
+package main
+
+import (
+	"testing"
+
+	"github.com/openshift/ci-tools/pkg/testhelper"
+)
+
+func TestUpdateDockerfile(t *testing.T) {
+	testCases := []struct {
+		name           string
+		dockerfile     []byte
+		config         ocpImageConfig
+		expectedErrMsg string
+		expectUpdate   bool
+	}{
+		{
+			name: "No changes",
+			dockerfile: []byte(`# This dockerfile is used for building for OpenShift
+FROM rhel-8-golang as rhel8
+ADD . /go/src/github.com/dougbtv/whereabouts
+WORKDIR /go/src/github.com/dougbtv/whereabouts
+ENV CGO_ENABLED=1
+ENV GO111MODULE=on
+ENV VERSION=rhel8 COMMIT=unset
+RUN go build -mod vendor -o bin/whereabouts cmd/whereabouts.go
+WORKDIR /
+
+FROM golang as rhel7
+ADD . /go/src/github.com/dougbtv/whereabouts
+WORKDIR /go/src/github.com/dougbtv/whereabouts
+ENV CGO_ENABLED=1
+ENV GO111MODULE=on
+RUN go build -mod vendor -o bin/whereabouts cmd/whereabouts.go
+WORKDIR /
+
+FROM rhel
+RUN mkdir -p /usr/src/whereabouts/images && \
+       mkdir -p /usr/src/whereabouts/bin && \
+       mkdir -p /usr/src/whereabouts/rhel7/bin && \
+       mkdir -p /usr/src/whereabouts/rhel8/bin
+COPY --from=rhel7 /go/src/github.com/dougbtv/whereabouts/bin/whereabouts /usr/src/whereabouts/rhel7/bin
+COPY --from=rhel7 /go/src/github.com/dougbtv/whereabouts/bin/whereabouts /usr/src/whereabouts/bin
+COPY --from=rhel8 /go/src/github.com/dougbtv/whereabouts/bin/whereabouts /usr/src/whereabouts/rhel8/bin
+
+LABEL io.k8s.display-name="Whereabouts CNI" \
+      io.k8s.description="This is a component of OpenShift Container Platform and provides a cluster-wide IPAM CNI plugin." \
+      io.openshift.tags="openshift" \
+      maintainer="CTO Networking <nfvpe-container@redhat.com>"`),
+			config: ocpImageConfig{From: ocpImageConfigFrom{
+				Builder: []ocpImageConfigFromStream{{"rhel-8-golang"}, {"golang"}},
+				Stream:  "rhel",
+			}},
+		},
+		{
+			name: "Dockerfile gets updated, comment preceeding directive",
+			dockerfile: []byte(`# This dockerfile is used for building for OpenShift
+FROM openshift/origin-release:rhel-8-golang-1.12 as rhel8
+FROM something
+`),
+			config: ocpImageConfig{From: ocpImageConfigFrom{
+				Builder: []ocpImageConfigFromStream{{"replaced"}},
+				Stream:  "replacement-2",
+			}},
+			expectUpdate: true,
+		},
+		{
+			name: "Dockerfile gets updated, no comment preceeding directive",
+			dockerfile: []byte(`FROM openshift/origin-release:rhel-8-golang-1.12 as rhel8
+FROM something
```

Comment:
Is it always expected to have just two FROMs and for them to always be ordered as builder first and stream second?

Reply:
There must be at least one from but there might be arbitrary many additional builder images referenced